fluent/protocol/forward_message_test.go

```diff
@@ -0,0 +1,119 @@
+package protocol_test
+
+import (
+	"bytes"
+	"io/ioutil"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/tinylib/msgp/msgp"
+
+	. "github.ibm.com/Observability/fluent-forward-go/fluent/protocol"
+)
+
+var _ = Describe("ForwardMessage", func() {
+	var (
+		fwdmsg ForwardMessage
+	)
+
+	BeforeEach(func() {
+		bits, err := ioutil.ReadFile("protocolfakes/forwarded_records.msgpack.bin")
+		Expect(err).ToNot(HaveOccurred())
+		_, err = fwdmsg.UnmarshalMsg(bits)
+		Expect(err).NotTo(HaveOccurred())
+		fwdmsg = ForwardMessage{
+			Tag: "foo",
+			Entries: []EntryExt{
+				{
+					Timestamp: EventTime{time.Now()},
+					Record: map[string]interface{}{
+						"foo":    "bar",
+						"george": "jungle",
+					},
+				},
+				{
+					Timestamp: EventTime{time.Now()},
+					Record: map[string]interface{}{
+						"foo":    "kablooie",
+						"george": "frank",
+					},
+				},
+			},
+		}
+	})
+
+	Describe("Unmarshaling", func() {
+		testMarshalling := func(msg ForwardMessage, opts *MessageOptions) {
+			msg.Options = opts
+			b, err := msg.MarshalMsg(nil)
+			Expect(err).NotTo(HaveOccurred())
+
+			var unmfwd ForwardMessage
+			_, err = unmfwd.UnmarshalMsg(b)
+			Expect(err).NotTo(HaveOccurred())
+
+			if opts == nil {
+				Expect(unmfwd.Options).To(BeNil())
+			} else {
+				Expect(unmfwd.Options).ToNot(BeNil())
+			}
+			Expect(unmfwd.Tag).To(Equal("foo"))
+			Expect(unmfwd.Entries[0].Timestamp.Time.Equal(msg.Entries[0].Timestamp.Time)).To(BeTrue())
+			Expect(unmfwd.Entries[0].Record).To(HaveKeyWithValue("foo", "bar"))
+			Expect(unmfwd.Entries[0].Record).To(HaveKeyWithValue("george", "jungle"))
+			Expect(unmfwd.Entries[1].Timestamp.Time.Equal(msg.Entries[1].Timestamp.Time)).To(BeTrue())
+			Expect(unmfwd.Entries[1].Record).To(HaveKeyWithValue("foo", "kablooie"))
+			Expect(unmfwd.Entries[1].Record).To(HaveKeyWithValue("george", "frank"))
+		}
+
+		It("Marshals and unmarshals correctly", func() {
+			// Uncomment after adding custom encoding
+			// testMarshalling(fwdmsg, nil)
```

Comment:
@ericmillin - did we need to uncomment and execute this test, too?  your comment implies that we do.

Reply:
Custom encoding is a TODO. I can make that explicit in the comment or remove it.

Reply:
As long as the commenting is intentional, I'm fine.  Just wanted to check.